htdocs/api/v0.0.3-dev/candidates/visits/dicoms/Dicom.php

```diff
@@ -0,0 +1,147 @@
+<?php
+/**
+ * Handles API requests for the candidate's visit dicom tar file
+ *
+ * PHP Version 5.5+
+ *
+ * @category Main
+ * @package  API
+ * @author   Mouna Safi-Harab <mouna.safiharab@gmail.com>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+namespace Loris\API\Candidates\Candidate\Visit\Dicoms;
+require_once __DIR__ . '/../../Visit.php';
+/**
+ * Handles API requests for the candidate's visit dicom tar file. Extends
+ * Visit so that the constructor will validate the candidate
+ * and visit_label portion of URL automatically.
+ *
+ * @category Main
+ * @package  API
+ * @author   Mouna Safi-Harab <mouna.safiharab@gmail.com>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+class Dicom extends \Loris\API\Candidates\Candidate\Visit
+{
+    /**
+     * Construct a Dicom class object to output the candidate visits dicom tar file
+     *
+     * @param string $method     The method of the HTTP request
+     * @param string $CandID     The CandID to be serialized
+     * @param string $VisitLabel The visit label to be serialized
+     * @param string $Tarname    The Tar name to be retrieved
+     */
+    public function __construct($method, $CandID, $VisitLabel, $Tarname)
+    {
+        $requestDelegationCascade = $this->AutoHandleRequestDelegation;
+
+        $this->AutoHandleRequestDelegation = false;
+
+        if (empty($this->AllowedMethods)) {
+            $this->AllowedMethods = ['GET'];
+        }
+        $this->CandID     = $CandID;
+        $this->VisitLabel = $VisitLabel;
+        $this->Tarname    = $Tarname;
+
+        parent::__construct($method, $CandID, $VisitLabel);
+
+        $results =  $this->getFullPath();
+        if (empty($results)) {
+            $this->header("HTTP/1.1 404 Not Found");
+            $this->error("File not found");
+            $this->safeExit(0);
+        }
+
+        if ($requestDelegationCascade) {
+            $this->handleRequest();
+        }
+
+    }
+
+    /**
+     * Handles a GET request
+     *
+     * @return none, but send the file content to stdout
+     */
+    public function handleGET()
+    {
+        $fullDir = $this->getFullPath();
+        $fp      = fopen($fullDir, "r");
+        if ($fp !== false) {
+            $this->Header("Content-Type: application/x-tar");
+            $this->Header('Content-Length: '.filesize($fullDir));
+            $this->Header('Content-Disposition: filename='.$this->Tarname);
```

Comment:
Missing "attachment" in the header.

Reply:
@MounaSafiHarab for point 2. I agree, it would make more sense to label it with the CandID_PSCID_VL. The only question, do we want to keep a trace of what was the tarname? If so, then maybe it could be downloaded as the tarname appended to the CandID_PSCID_VL? If not, simply CandID_PSCID_VL would work.

Hope that makes sense!

Reply:
Do the minc files have a `Content-Disposition` header?

The values for Content-Disposition headers are "attachment" or "inline" for determining if the URL is intended to be downloaded or displayed inline in a page. "filename" is an argument for attachment (since it doesn't make sense for inline..), which is why I say it's missing "attachment".. filename by itself isn't a valid value for the headers without specifying what the disposition is.

I think the filename should be the same as the URL endpoint, whatever that is.

Reply:
sorry i have another dumb question but what's the difference between doing `$this->Header(...)` and `$this->header(...)`
is `Header()` just a wrapper?

Reply:
@davidblader ; I am assuming the question is addressed to @driusan , as I do not know (the changes all come from the Image

@driusan @cmadjar 
if the filename is to be kept as the URL, then it is the name I currently have 
@cmadjar if we were to download files with a different filename, it should come in a different pull request, simply because the changes to `get_file.php` will impact many other places as well (DICOM archive, Media, Imaging Browser potentially)... but I like your suggestion of appending as opposed to changing the original filename; might be the best compromise?

Reply:
> Do the minc files have a Content-Disposition header?

yes they do. A big chunk of this script comes from the MINC counterpart.

Reply:
@davidblader Function names in `PHP` are case insensitive. There's no difference between the two lines you pasted.